packages/http-client-csharp/generator/README.md

```diff
@@ -1,11 +1,21 @@
 # Microsoft TypeSpec Generator
 
-The **Microsoft TypeSpec Generator** tool generates client libraries for accessing RESTful web services.
+The **Microsoft TypeSpec Generator** is a sophisticated code generation framework that transforms TypeSpec API definitions into strongly-typed C# client libraries for accessing RESTful web services. The generator provides a modular, extensible architecture that supports multiple output formats and customization scenarios.
 
 ## Table of Contents
 
+- [Architecture Overview](#architecture-overview)
+- [Core Components](#core-components)
```

Comment:
I'm a little confused by this ToC since it seems to be the ToC for the architecture doc.